### ISSUE TITLE
Decouple env var from apprunner.yaml

### ISSUE THREAD

**Issue by robmarkcole** [aws/apprunner-roadmap#29]
**Community Note**
* Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do * not help prioritize the request
If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Have an `apprunner.yaml` file with some of the core info such as the language runtime, etc and decoupling the env variable for convenience/secrecy which can be set independently via the console.

**Describe alternatives you've considered**
Discussion on https://github.com/aws-containers/hello-app-runner/issues/4

**Additional context**
NA

**Attachments**
None

**Comment by toricls** [comment]
+1. The ability to define env vars in apprunner.yaml is useful but it would be helpful to me if it allows to override env vars declared in apprunner.yaml by the API or the management console.

**Comment by mattpmoreno** [comment]
has there been any movement on this issue? It would seem pretty straightforward that env variable be defined in AppRunner service page AND/OR in the yaml file...

**Comment by UnderpantsGnome** [comment]
I have no idea when this was added, but you can add up to 50 ENV vars or secrets via the UI or CLI.

![Screenshot 2024-05-18 at 6 51 46 PM](https://github.com/aws/apprunner-roadmap/assets/17270/a8943ba9-3465-4272-8310-5c0989927390)

**Comment by adamauckland** [comment]
> I have no idea when this was added, but you can add up to 50 ENV vars or secrets via the UI or CLI.
> 
> 
> 
> ![Screenshot 2024-05-18 at 6 51 46 PM](https://github.com/aws/apprunner-roadmap/assets/17270/a8943ba9-3465-4272-8310-5c0989927390)
> 
> 
> 
> 

Thanks, @UnderpantsGnome , you are right, but only if you don't use apprunner.yaml and configure an OCI and build manually, which defeats the point of AppRunner which is a rapid, "serverless" solution.

If you use apprunner.yaml, there is no way to add extra environment variables!

**Comment by UnderpantsGnome** [comment]
@adamauckland apologies, we're not using `apprunner.yml` in any of our services (we deploy from ECR), and that seems to be the one step that stops you from adding ENV vars. :(

**Comment by adamauckland** [comment]
> @adamauckland apologies, we're not using `apprunner.yml` in any of our services (we deploy from ECR), and that seems to be the one step that stops you from adding ENV vars. :(

Yeah, I'm using docker compose in development, so maybe I just need to push the main container up, just seems like I shouldn't have to! 😆 like nobody thought "Hang on, people might want to deploy this to different environments with different settings"

**Comment by GitHunter0** [comment]
App Runner is a really great product, I just wish it would have a way more active development to fix issues like this...
